Owncloud iOs Client/Login/Keychain/OCKeychain.m

```diff
@@ -23,25 +23,62 @@
 @implementation OCKeychain
 
 
-+(BOOL)setCredentialsById:(NSString *)idUser withUsername:(NSString *)userName andPassword:(NSString *)password{
++(BOOL)setCredentialsById:(NSString *)userId withUsername:(NSString *)userName andPassword:(NSString *)password{
+    
+    BOOL output = NO;
+    
+    [self setCredentialsById:userId withUsername:userName andData:[password dataUsingEncoding:NSUTF8StringEncoding]];
+    
+//    NSMutableDictionary *keychainItem = [NSMutableDictionary dictionary];
+//    
+//    [keychainItem setObject:(__bridge id)(kSecClassGenericPassword) forKey:(__bridge id)kSecClass];
+//    [keychainItem setObject:(__bridge id)(kSecAttrAccessibleAfterFirstUnlock) forKey:(__bridge id)kSecAttrAccessible];
+//    [keychainItem setObject:idUser forKey:(__bridge id)kSecAttrAccount];
+//    [keychainItem setObject:userName forKey:(__bridge id)kSecAttrDescription];
+//    [keychainItem setObject:[UtilsUrls getFullBundleSecurityGroup] forKey:(__bridge id)kSecAttrAccessGroup];
+//
+//    
+//    OSStatus stsExist = SecItemCopyMatching((__bridge CFDictionaryRef)keychainItem, NULL);
+//    
+//    if(stsExist == errSecSuccess) {
+//        NSLog(@"Unable add item with id =%@ error",idUser);
+//    }else {
+//        [keychainItem setObject:[password dataUsingEncoding:NSUTF8StringEncoding] forKey:(__bridge id)kSecValueData];
+//        OSStatus stsAdd = SecItemAdd((__bridge CFDictionaryRef)keychainItem, NULL);
+//        
+//        NSLog(@"(setCredentials)Error Code: %d", (int)stsAdd);
+//        if (stsAdd == errSecSuccess) {
+//            output = YES;
+//        }
+//        
+//    }
+    
+    return output;
+}
+
+
++(BOOL)setCredentialsById:(NSString *)userId
+             withUsername:(NSString *)userName
+              andData:(NSData *)data {
     
     BOOL output = NO;
     
     NSMutableDictionary *keychainItem = [NSMutableDictionary dictionary];
     
-    [keychainItem setObject:(__bridge id)(kSecClassGenericPassword) forKey:(__bridge id)kSecClass];
```

Comment:
Doesn't matter, just I want to use some key that are by default in the kSecClass kSecClassInsternetPassword as kSecAttrAuthenticationType. This was first approach but after I store all as data. This keychain method maybe not be the final, it is not in use yet. At this moment we can keep using kSecClassGenericPassword because I no longer use the other attributes